### PR TITLE
chore: refactor runtimes code

### PR DESCRIPTION
**Pull request by ntn-x2**
WIP. Fixes https://github.com/KILTprotocol/ticket/issues/3672. Built on top of https://github.com/KILTprotocol/kilt-node/pull/790.

## Checklist
- [x] Review Peregrine code
- [x] Ask for review
- [ ] Apply to Spiritnet code

### CODE REVIEW

File: runtimes/kestrel/src/lib.rs [hunk 1]
```diff
@@ -676,9 +677,9 @@ impl pallet_multisig::Config for Runtime {
 	type RuntimeEvent = RuntimeEvent;
 	type RuntimeCall = RuntimeCall;
 	type Currency = Balances;
-	type DepositBase = constants::multisig::DepositBase;
-	type DepositFactor = constants::multisig::DepositFactor;
-	type MaxSignatories = constants::multisig::MaxSignitors;
+	type DepositBase = ConstU128<{ constants::multisig::DEPOSIT_BASE }>;
+	type DepositFactor = ConstU128<{ constants::multisig::DEPOSIT_FACTOR }>;
+	type MaxSignatories = ConstU32<{ constants::multisig::MAX_SIGNITORS }>;
```
Comment (Ad96el) [inline runtimes/kestrel/src/lib.rs:682]: Why these changes? Do you think it is more declarative?

File: runtimes/peregrine/src/benchmarks/mod.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use frame_benchmarking::define_benchmarks;
```
Comment (Ad96el) [inline runtimes/peregrine/src/benchmarks/mod.rs:19]: The whole impl frame_benchmarking::Benchmark<Block> for Runtime
is missing for peregrine. Do we not still need that? 

Also the benchmark implementation for XCM is missing

File: runtimes/peregrine/src/kilt.rs
Comment (Ad96el) [inline runtimes/peregrine/src/kilt.rs:1]: Not a super big fan of this split. I would rather like to see an identity section. 

Pallet such as the parachain-staking are for me more system pallets. Same as the switch pallet. Dip is more an identy pallet

File: runtimes/peregrine/src/governance.rs [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+	type RuntimeOrigin = RuntimeOrigin;
+	type PalletsOrigin = OriginCaller;
+	type RuntimeCall = RuntimeCall;
+	type MaximumWeight = MaximumSchedulerWeight;
+	type ScheduleOrigin = RootOrCollectiveProportion<CouncilCollective, 1, 2>;
+	type MaxScheduledPerBlock = ConstU32<50>;
+	type WeightInfo = weights::pallet_scheduler::WeightInfo<Runtime>;
+	type OriginPrivilegeCmp = OriginPrivilegeCmp;
+	type Preimages = Preimage;
+}
```
Comment (Ad96el) [inline runtimes/peregrine/src/governance.rs:292]: Is this not more a system pallet?

File: runtimes/peregrine/src/kilt.rs [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+	#[cfg(feature = "runtime-benchmarks")]
+	type BenchmarkHelper = crate::benchmarks::asset_switch::CreateFungibleForAssetSwitchPool1;
+}
+
+impl pallet_migration::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Currency = Balances;
+	type MaxMigrationsPerPallet = constants::pallet_migration::MaxMigrationsPerPallet;
+	type WeightInfo = weights::pallet_migration::WeightInfo<Runtime>;
+}
```
Comment (Ad96el) [inline runtimes/peregrine/src/kilt.rs:357]: Lazy migration pallets should, maybe also live in the migration section.

File: runtimes/common/src/constants.rs [hunk 5]
```diff
@@ -363,14 +363,19 @@ pub mod governance {
...
 	use super::*;
 
-	parameter_types! {
-		pub const MaxSignitors: u32 = 64;
-		pub const DepositBase: Balance = DEPOSIT_STORAGE_ITEM;
-		pub const DepositFactor: Balance = DEPOSIT_STORAGE_BYTE;
-	}
+	pub const MAX_SIGNITORS: u32 = 64;
+	pub const DEPOSIT_BASE: Balance = DEPOSIT_STORAGE_ITEM;
+	pub const DEPOSIT_FACTOR: Balance = DEPOSIT_STORAGE_BYTE;
```
Comment (Ad96el) [inline runtimes/common/src/constants.rs:378]: Is there a reason to drop the parameter_types macro and using the Const?

File: runtimes/kestrel/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ntn-x2) [reply runtimes/kestrel/src/lib.rs:682]: Yeah, I like this approach more that relies on well-known Rust primitives like `const`s and const generic, rather than some Substrate macro. What do you think?

File: runtimes/peregrine/src/kilt.rs
Comment (ntn-x2) [reply runtimes/peregrine/src/kilt.rs:1]: Mhh this we could argue about. The parachain staking is a KILT pallet, and KILT-specific. Not an identity feature per se tho. I would not want to put it in system, which is envisioned, in my opinion, as a collection of the configs that are common to most parachains. But I don't have a super strong opinion on this. Also I wanted to put stuff like the multisig pallet and something else in a different module since they're not critical system components, but I ended up putting them there for lack of a better alternative.

File: runtimes/common/src/constants.rs (hunk 5 above)
Comment (ntn-x2) [reply runtimes/common/src/constants.rs:378]: As I said above, this is more a matter of a personal preference. It uses the Rust primitives types instead of a macro. I probably should not have done this refactoring in this PR as well, so I will probably revert these changes and migrate everything to the new syntax in a different PR, if you agree this is better.

File: runtimes/peregrine/src/benchmarks/mod.rs (hunk 2 above)
Comment (ntn-x2) [reply runtimes/peregrine/src/benchmarks/mod.rs]: Ops! Big catch, thank you! I implemented the required functions in the benchmarking module and call into them from the runtime API impl module: https://github.com/KILTprotocol/kilt-node/pull/797/commits/e5c2f73f31dd17f2e6c44c287fba7e66c429cc12.

File: runtimes/common/src/constants.rs (hunk 5 above)
Comment (ntn-x2) [reply runtimes/common/src/constants.rs]: I reverted the changes in here: https://github.com/KILTprotocol/kilt-node/pull/797/commits/08e3236e09db6ea4974bb3729551d4ef1f106993. I will change all of them if you agree in a separate PR.

File: runtimes/kestrel/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ntn-x2) [reply runtimes/kestrel/src/lib.rs]: I reverted the changes in here: https://github.com/KILTprotocol/kilt-node/pull/797/commits/08e3236e09db6ea4974bb3729551d4ef1f106993. I will change all of them if you agree in a separate PR.

File: runtimes/peregrine/src/governance.rs (hunk 3 above)
Comment (ntn-x2) [reply runtimes/peregrine/src/governance.rs]: Moved it into the system module: https://github.com/KILTprotocol/kilt-node/pull/797/commits/bf159368bb2127655073a6f5e5849f1fe6781285.

File: runtimes/peregrine/src/kilt.rs (hunk 4 above)
Comment (ntn-x2) [reply runtimes/peregrine/src/kilt.rs]: Moved it into the migration module, good suggestion: https://github.com/KILTprotocol/kilt-node/pull/797/commits/24c894ac807fd1d4867339b6afb69900accafcad.

File: runtimes/peregrine/src/benchmarks/mod.rs (hunk 2 above)
Comment (Ad96el) [reply runtimes/peregrine/src/benchmarks/mod.rs]: Did you try running the benchmarks for the pallet? That would be a good indicator of whether we introduced any nasty bugs during the refactor.

File: runtimes/peregrine/src/kilt.rs
Comment (Ad96el) [reply runtimes/peregrine/src/kilt.rs:1]: don’t have a strong opinion on this either. I think we can stick with the current approach, but perhaps we could refine it slightly. Instead of a single kilt.rs file, we could introduce a kilt module to organize all custom pallets. For example:
 
```
 kilt
-> mod.rs // Contains parachain staking & switch, or these can go into a separate file like funds.rs
-> dip.rs
-> did.rs // Not sure if this is the best name, but I think w3n, credentials, and linking all fall under DID functionality.
```

Do you think this nested approach might be too confusing?

Feel free to disregard this suggestion if it doesn’t resonate. I just want to maintain consistency, and having a separate dip module is slightly bothering me.

File: runtimes/kestrel/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ad96el) [reply runtimes/kestrel/src/lib.rs]: I think being more sticky to rust primitives is better than substrate

[the remark this answers is not in the document]
Comment (ntn-x2) [reply runtimes/kestrel/src/lib.rs]: Will do!

File: runtimes/peregrine/src/benchmarks/mod.rs (hunk 2 above)
Comment (ntn-x2) [reply runtimes/peregrine/src/benchmarks/mod.rs]: Ehm... I did now, and found out the issue can be resolved only by actually mocking the storage for the prime member. I did it here: https://github.com/KILTprotocol/kilt-node/pull/797/commits/f2c2d4629c53b4454bac23360a8dcdab38f0093e, and refactored it into its own module in the benchmarking module here: https://github.com/KILTprotocol/kilt-node/pull/797/commits/be50327e707d12a35e36cf16a7136789b68b9630

File: runtimes/peregrine/src/kilt.rs
Comment (Ad96el) [reply runtimes/peregrine/src/kilt.rs:1]: I think it's very nice :+1:

File: runtimes/peregrine/src/benchmarks/mod.rs (hunk 2 above)
Comment (Ad96el) [reply runtimes/peregrine/src/benchmarks/mod.rs]: And all other benchmarks did work? No strange spikes in the calculated weights?

Comment (ntn-x2) [reply runtimes/peregrine/src/benchmarks/mod.rs]: Did not run them on the benchmarking server. That is something we will do during the release. As far as I am concerned, all benchmarks worked, and no `verify` failed, so I'm good with that.

**Review by Ad96el — DISMISSED** [review]
LGTM!
